### PR TITLE
[Enhancement] Add simdjson_iterate_batch to set window batch size for simdjson::iterate_many

### PR DESCRIPTION
**Pull request by rickif**
Why I'm doing:
Now, the batch window is set as the payload size of json input. The simdjson parser would return error `simdjson::CAPACITY` when the document stream size is more than 4GB.
https://github.com/simdjson/simdjson/blob/1a195623a538d3ad1e9ae54b552655b696e57ac2/doc/basics.md#L1224
> The `iterate_many` function can also take an optional parameter `size_t batch_size` which defines the window processing size. It is set by default to a large value (`1000000` corresponding to 1 MB). None of your JSON documents should exceed this window size, or else you will get  the error `simdjson::CAPACITY`. You cannot set this window size larger than 4 GB: you will get  the error `simdjson::CAPACITY`. The smaller the window size is, the less memory the function will use. Setting the window size too small (e.g., less than 100 kB) may also impact performance negatively. Leaving it to 1 MB is expected to be a good choice, unless you have some larger documents.

What I'm doing:
This PR makes the `batch_size` of iterate_many configurable.  An error `The document is too big to fit in the parser's window. Please increase the parser's windows size by setting new BE config simdjson_iterate_many`  will be returned when the json document size is greater than this setting. The user can modify the setting to eliminate this error.
The default value of `simdjson_iterate_batch` is 1GB, and the maximum value is 4GB.

With this PR, users can parse ndjson format files that are more than 4GB but no more than `simdjson_iterate_batch` (default 1GB)  for a single json document.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [x] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -1145,4 +1145,7 @@ CONF_mInt64(lake_local_pk_index_unused_threshold_seconds, "86400"); // 1 day
 
 CONF_mBool(lake_enable_vertical_compaction_fill_data_cache, "false");
 
+// simdjson_iterate_batch sets the maximum size of single json document. Default 1GB.
+CONF_mInt64(simdjson_iterate_batch, "1073741824");
```
Comment (wyb) [inline be/src/common/config.h:1149]:
```suggestion
CONF_mInt64(simdjson_iterate_batch_size, "1073741824");
```

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by rickif** [comment]
@mergifyio rebase

**Review by wyb — APPROVED** [review]

**Pull request closed without merging**